packages/aggregator/src/http/state.rs

```diff
@@ -76,25 +77,35 @@ impl HttpState {
     #[instrument(skip(config, metrics))]
     pub fn new(
         config: Config,
+        chain_configs: Arc<RwLock<ChainConfigs>>,
         metrics: utils::telemetry::AggregatorMetrics,
     ) -> AggregatorResult<Self> {
-        Self::new_with_engine(config, metrics)
+        Self::new_with_engine(config, chain_configs, metrics)
     }
 
     #[instrument(skip(config, metrics))]
     pub fn new_with_engine(
         config: Config,
+        chain_configs: Arc<RwLock<ChainConfigs>>,
         metrics: utils::telemetry::AggregatorMetrics,
     ) -> AggregatorResult<Self> {
         tracing::info!("Creating file storage at: {:?}", config.data);
         let file_storage = FileStorage::new(&config.data)?;
         let ca_storage = Arc::new(file_storage);
         let storage = RedbStorage::new(config.data.join("db"))?;
         let evm_clients = Arc::new(RwLock::new(HashMap::new()));
+        let engine_chain_configs = if config.dev_endpoints_enabled {
```

Comment:
I don't _love_ this... not worth blocking the PR on since it's logically fine, but it's a bit hairy... can we not make the config itself have its chain configs as `Arc<RwLock<ChainConfigs>>` ?